src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,95 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    public static final int DEFAULT_CAPACITY = 16;
+    public static final float LOAD_FACTOR = 0.75f;
+
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        this.table = new Node[DEFAULT_CAPACITY];
+        this.threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        int hash = hash(key);
+        int index = indexFor(hash, table.length);
 
+        for (Node<K, V> node = table[index]; node != null; node = node.next) {
+            if (node.hash == hash && (node.key == key || (key != null && key.equals(node.key)))) {
+                node.value = value;
+                return;
+            }
+        }
+
+        addNode(hash, key, value, index);
     }
 
     @Override
     public V getValue(K key) {
+        int hash = hash(key);
+        int index = indexFor(hash, table.length);
+
+        for (Node<K, V> node = table[index]; node != null; node = node.next) {
+            if (node.hash == hash && (node.key == key || (key != null && key.equals(node.key)))) {
+                return node.value;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
```

Comment:
The `getSize` method should not be public if it is only intended to be used within the class. If it is part of the `MyMap` interface and should be public, then disregard this comment.